### PR TITLE
FMC : Log PCR extensions to the PCR_Log

### PR DESCRIPTION
**Pull request by rusty1968**
FMC appends to the PCR log in DCCM, to enhance testability.

### CODE REVIEW

File: fmc/src/flow/pcr.rs [hunk 1]
```diff
@@ -23,9 +23,30 @@ Note:
...
+impl PcrExtender<'_> {
+    fn extend(
+        &mut self,
+        pcr_id: PcrId,
+        bytes: &[u8],
+        pcr_entry_id: PcrLogEntryId,
+    ) -> CaliptraResult<()> {
+        self.pcr_bank
+            .extend_pcr(PcrId::PcrId0, self.sha384, bytes)?;
+        log_pcr(pcr_entry_id, pcr_id, bytes)
```
Comment (bluegate010) [inline fmc/src/flow/pcr.rs:47]: With https://github.com/chipsalliance/caliptra-sw/pull/600, we could adjust this logic a bit such that we only create one log when extending into both PCRs. (This wouldn't be strictly necessary, we could keep creating one log for each extend operation, but it's more space efficient if we can group them together).

File: fmc/src/flow/pcr.rs [hunk 2]
```diff
@@ -56,16 +77,62 @@ pub fn extend_journey_pcr(env: &mut FmcEnv, hand_off: &HandOff) -> CaliptraResul
...
+    };
+    pcr_log_entry.pcr_data.as_bytes_mut()[..data.len()].copy_from_slice(data);
+
+    let dst: &mut [PcrLogEntry] = unsafe {
+        let ptr = PCR_LOG_ORG as *mut PcrLogEntry;
+        core::slice::from_raw_parts_mut(ptr, PCR_LOG_SIZE / core::mem::size_of::<PcrLogEntry>())
+    };
+
+    // Store the log entry.
+    dst[pcr_entry_id as usize - 1] = pcr_log_entry;
```
Comment (bluegate010) [inline fmc/src/flow/pcr.rs:135]: This current logic will leave holes in the log buffer if ROM and FMC do not agree on how many extensions should have occurred. https://github.com/chipsalliance/caliptra-sw/pull/482 brings in a log entry counter, which ROM initializes to zero, and which FMC can initialize by walking the log space until it finds an invalid entry. Or, could have ROM put the entry count in FHT.

File: fmc/src/flow/pcr.rs (hunk 1 above)
Comment (rusty1968) [reply fmc/src/flow/pcr.rs:47]: Can you make that code common?

File: fmc/src/flow/pcr.rs (hunk 2 above)
Comment (FerralCoder) [reply fmc/src/flow/pcr.rs:135]: I think it would make the most sense for ROM to fill in this count, and for FMC to update the count

**Pull request closed without merging**